### ISSUE TITLE
strict mode: undeclared commodity "AUTO"

### ISSUE THREAD

**Issue by Flimm** [simonmichael/hledger#1419]
When in strict mode, hledger does not allow you to use automatically inferred posting prices. For example, take this file:

```
commodity USD 1 000.00
account assets
account revenues

2020/01/01 test
    assets  USD 100
    revenues
```

You would expect that this would parse correctly in strict mode, since all the accounts and the commodities have been declared. However, it doesn't:

```
$ hledger -f bla.journal balance -s
hledger: 
strict mode: undeclared commodity "AUTO"
in transaction at: "example/bla.journal" (lines 5-7)
```

**Comment by simonmichael** [comment]
Thanks for the bug report.

**Comment by simonmichael** [comment]
Opening as a reminder that this needs a bugfix release.

**Comment by simonmichael** [comment]
Closing to reduce open loops. This will finally be fixed by 1.20.3, coming soon.
